source/_integrations/homekit_controller.markdown

```diff
@@ -182,3 +182,7 @@ In these cases, HomeKit Controller will skip polling to avoid a buildup of back
 ### I can't see any events generated for "stateless" accessories
 
 This is expected. The only way to use stateless accessories like some doorbells, buttons or remotes with Home Assistant is through device automations. Home Assistant doesn't create duplicate events for device automation triggers, so for example you won't be able to watch them with the events developer tools.
+
+### Home Assistant can't see my homebridge device (≥2021.11.x)
+
+In your Homebridge settings/config, make sure you are using `ciao` and not `Bonjour-HAP`. `Bonjour-HAP` is no longer recommended by `homebridge` and is considered broken/unsupported. Using `Bonjour-HAP` will result in previously working devices/entities to become unavailable, and will result in a `Config flow not found` error prompt if attempting to add new devices.
```

Comment:
I realise it might be helpful for some types of users, but I'm reluctant to include the 2nd sentence. Thats what happens right now and in your environment. That's problematic for a few reasosn:

* How the problem manifests might change going forwards in later HA versions
* How it manifests in other environments might be different to how it manifests in yours
* This is valid troubleshooting advice for any Homebridge/homekit_controller discovery problem. This sentence might make someone think the advice does not apply to them.

Because we haven't recreated it in another environment we also can't say with 100% certainty which parts were caused directly by Bonjour-HAP too.

What I want to avoid is information with a shelf life. Advising users to switch to `ciao` doesn't have a shelf life, or at least it has a longer shelf life, if that makes sense. It's a good troubleshooting step for ANY HomeBridge issue.

Reply:
Got it, all makes sense